### PR TITLE
Fixes VSTS Bug 961599: [Feedback] Visual Studio crashes/exists when

### PR DESCRIPTION
**Pull request by mkrueger**
attempting to commit to Git

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/961599

MessageService was causing a stack Overflow it could add a dialog as
child to itself. However teh git commit dialog should set itself as
parent for the git config dialog.

### CODE REVIEW

**Comment by mkrueger** [comment]
@monojenkins backport release-8.2

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/CommitDialogExtension.cs [hunk 1]
```diff
@@ -6,8 +6,9 @@ namespace MonoDevelop.VersionControl
 	/// <summary>
 	/// Base class for commit dialog extensions.
 	/// </summary>
-	public class CommitDialogExtension: Gtk.EventBox
+	public class CommitDialogExtension : Gtk.EventBox
 	{
+		public VersionControl.Dialogs.CommitDialog { get; set; }
```
Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/CommitDialogExtension.cs]: Why public?

**Comment by mkrueger** [comment]
@monojenkins backport release-8.2

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/CommitDialogExtension.cs (hunk 1 above)
Comment (mkrueger) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/CommitDialogExtension.cs]: Was too late yesterday - I changed it to internal. Needed to access that in another project.

But y we shouldn't change the public API. IVT all the way :)

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/MessageService.cs [hunk 2]
```diff
@@ -330,7 +330,7 @@ public static int RunCustomDialog (Dialog dlg, Window parent)
 				if (dialog.TransientFor != null)
 					parent = dialog.TransientFor;
 				else
-					parent = IdeServices.DesktopService.GetFocusedTopLevelWindow ();
```
Comment (sevoku) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/MessageService.cs]: This is not right, `GetFocusedTopLevelWindow ()` has an additional check to not parent on pads, while `GetParentForModalWindow ()` just uses the top window (which might be also a pad) which is not necessarily focused. Actually we should deprecate `GetParentForModalWindow ()`, but I need to review that ...

Comment (sevoku) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/MessageService.cs]: The right fix would be to check that `parent != dialog`, if the window is the same one throw an InvalidOperationException

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/MessageService.cs]: fixed.

**Comment by slluis** [comment]
@monojenkins backport release-8.2

**Comment by sevoku** [comment]
@monojenkins backport release-8.2

**Comment by sevoku** [comment]
@monojenkins backport release-8.3

**Pull request merged**